kernel/thread/mutex.c

```diff
@@ -133,6 +133,15 @@ int mutex_lock_timed(mutex_t *m, int timeout) {
             deadline = timer_ms_gettime64() + timeout;
 
         for(;;) {
+            if (m->holder->prio > thd_current->prio) {
```

Comment:
Would it make sense for this to be `>=` ? That wouldn't change the priority level, but what it would do is force it to be rescheduled to the front of the line for its priority level. So if you have 10 threads with the same priority in queue and No.10 in the queue is holding the mutex that No.1 needs, it'll get rescheduled to run before No.2 in the queue.

I don't think that'd be too disruptive or unfair to the scheduling, but I'm not 100% and it might just cause a bit too much extra overhead.

Reply:
Good idea.

Reply:
@QuzarDC Now that I see it in master, I'm pretty sure that we don't add it to the front of the line, but to the back of the line. `thd_add_to_runnable`'s last argument is named `front_of_line`, and we set it to 0.